### DISCUSSION TITLE
Is there a way to get a sensor for the Nth collection?

### DISCUSSION THREAD

**Discussion by Cadair** [mampfes/hacs_waste_collection_schedule#1099] [Q&A] [answered]
Before discovering this awesome plugin I had my sensors setup so that I had two sensors one for the next collection and one for the following collection, this let me easily show an ordered list of collections with the count of days until that collection:
![clipboard](https://github.com/mampfes/hacs_waste_collection_schedule/assets/1391051/689f8596-bbef-4d0e-be4a-1be040060b85)
I can't seem to figure out a way to reproduce this using this plugin?

**Comment by dt215git** [comment]
If you search for your current sensor in `Developer Tools` `States`,  does it show multiple collections/dates in the attributes?

**Reply by Cadair** [reply]
Yes it does. I guess I could do what I want with set of complex template sensors but I was hoping there would be a better way.

**Reply by dt215git** [reply]
Haven't tried it, but I guess the alternative could to be to have three identical sources, and use [customise](https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/installation.md#attributes-for-customize) to limit each one to a single collection type. Then you could have 3 sensors, using [source_index](https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/installation.md#configuring-sensors) to ensure you have one for each source. Would that get you closer to how your original sensors/cards were set up?

**Reply by Cadair** [reply]
It's the ordering of them that I am struggling with, how do I get a sensor which is always the second bin collection no matter what the type?

**Reply by dt215git** [reply]
I think that's probably going to need a template sensor. 
I've not spent much time using them, but playing around in `Settings` `Template` I ended up with building a list using the _Bin name_ and the _daysTo_ sensor attributes, reordering it and build a dict, then access the Nth entry when designing the card.

Something like this:

``` yaml
{% set ns = namespace(res=[]) %}
{% set my_list = [("Black", 5), ("Brown", 3), ("Green", 1)] %}
{% for k, v in my_list | sort(attribute='1') %}
  {% set ns.res = ns.res + [dict(bin = k, days = v)] %}
{% endfor %}

{{ ns.res }}

{{ ns.res[0] }} - {{ ns.res[0]["bin"] }} - {{ ns.res[0]["days"] }}
{{ ns.res[1] }} - {{ ns.res[1]["bin"] }} - {{ ns.res[1]["days"] }}
{{ ns.res[2] }} - {{ ns.res[2]["bin"] }} - {{ ns.res[2]["days"] }}
```

```
[{'bin': 'Green', 'days': 1}, {'bin': 'Brown', 'days': 3}, {'bin': 'Black', 'days': 5}]

{'bin': 'Green', 'days': 1} - Green - 1
{'bin': 'Blue', 'days': 3} - Brown - 3
{'bin': 'Black', 'days': 5} - Black - 5
```

Might need a bit of tweaking...

**Comment by 5ila5** [comment, marked as answer]
made a pull request (#1184) that would simplify this. But this is nothing I want to decide on my own so mampfes should have a look at it first.

UPDATE:

It's merged now and will be part of release 1.42.0 (or available now as master/git version)

[sensor documentation](https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/installation.md#configuring-sensors) (event_index) 

[example in faq "How do I configure sensors which show the first, second, third collection?"](https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/faq.md)

**Reply by Cadair** [reply]
That's awesome, thanks so much!!
